docs/source/federated-types/sharing-types.mdx

```diff
@@ -0,0 +1,404 @@
+
+The `@shareable` directive enables multiple subgraphs to resolve a particular field, or set of fields, on an Object type across multiple subgraphs.
+
+The `@shareable` directive is new in Federation 2. To enable it, you first need to add the following snippet to the top of **all of your subgraph files** to [opt-in to Federation 2](../federation-2/moving-to-federation-2/#opt-in-to-federation-2):
+
+```graphql
+extend schema
+    @link(url: "https://specs.apollo.dev/federation/v2.0",
+       import: ["@key", "@requires", "@shareable"])
+```
+
+Now you can apply the `@shareable` directive to the object types that you want to share across multiple subgraphs:
+
+<CodeColumns>
+
+```graphql{2}
+# Subgraph A
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+
+```graphql{2}
+# Subgraph B
+type Position @shareable {
+  x: Int!
+  y: Int!
+}
+```
+<></>
+
+</CodeColumns>
+
+> If a type or field is marked `@shareable` in _any_ subgraph, it must be marked `@shareable` or [`@external`](../federation-spec/#external) in _every_ subgraph that defines it. Otherwise, composition fails.
+
+The `@shareable` directive enables multiple subgraphs to resolve shared fields. So in the example above, both subgraphs A _and_ B can resolve the `Position.x` and `Position.y` fields.
+
+> ⚠️ **Important:** If multiple subgraphs can resolve a field, **make sure each subgraph's resolver for that field behaves identically.** Otherwise, queries might return inconsistent results depending on which subgraph resolves the field.
+
+Adding the `@shareable` directive to a type is a shortcut to sharing every field within that type. You can also share individual fields, like so:
+
+<CodeColumns>
+
+```graphql{3}
+# Subgraph A
+type Food {
+  name: String! @sharable
+  cost: Int!
+}
+```
+
+```graphql{3}
+# Subgraph B
+type Food {
+  name: String!  @sharable
+  inStock: Boolean!
+}
+```
+
+<></>
+
+</CodeColumns>
+
+If you'd like to learn how shared fields can differ across subgraphs, see [Varying shared fields](#varying-shared-fields).
+
+## Explicit and implicit type sharing
+
+### Objects
+
+By default, in Federation 2 schemas, Object type **fields** are not shared. This means that you cannot redefine an existing Object type's field without explicitly sharing that field.
+
+For example, let's say you'd like to define a `Product` type in two subgraphs:
+
+<p style="margin-bottom: 0">❌</p>
+
+<CodeColumns>
+
+```graphql{3, 4}
+# Subgraph A
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+```graphql{3, 4}
+# Subgraph B
+type Product {
+  name: String!
+  price: Int
+}
+```
+
+</CodeColumns>
+
+Attempting to compose these two subgraph schemas together will [**break composition**](./composition/#breaking-composition). The gateway doesn't know _which subgraph_ is responsible for resolving the `Product` type's fields.
+
+If we'd like both Subgraph A _and_ Subgraph B to be able to resolve the fields for `Product`, we can [explicitly share](#using-shareable) that type:
```

Comment:
"we'd like" -> "we want"

"fields for Product" -> "fields of Product"

"we can explicitly share that type" -> "we can explicitly mark those fields as shareable"